### DISCUSSION TITLE
parameter k in SAGEConv

### DISCUSSION THREAD

**Discussion by vivelakorea** [pyg-team/pytorch_geometric#8720] [Q&A] [answered]
Hi,

I've been investigating ways to aggregate neighbors for SAGEConv with the goal of encompassing not only the closest immediate neighbors but also those that are k steps away.

Although the parameter 'k' in SAGEConv lacked a clear explanation, I found that by specifying 'k=...' in the code, it executed successfully.

Could you kindly provide an explanation for the 'k' parameter in below code?

     from torch_geometric.nn import Linear, SAGEConv
     import torch.nn.functional as F

     class GNN(torch.nn.Module):

        def __init__(self,n,k):
            super(GNN, self).__init__()

            self.sage1 = SAGEConv(38,n,k=k,improved=True)
            self.out = Linear(n,1)

        def forward(self, data):
            x, edge_index= data.x, data.edge_index

            x = F.elu(self.sage1(x, edge_index))
            x = self.out(x)

            return x

**Comment by akihironitta** [comment, marked as answer]
As far as I understand the codebase, the kwarg `k` will be simply ignored. Instead, I'd stack `k` layers in the model to aggregate nodes that are `k` steps away from each seed node.

Out of curiosity, where have you found the usage of `SAGEConv(..., k=k)`?

**Reply by vivelakorea** [reply]
Actually one of my colleague recommended it, so I have no clue..
